source/common/router/router.h

```diff
@@ -138,6 +145,33 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     return callbacks_->connection();
   }
 
+  /**
+   * Set a pseudo-random cookie with the name @param key and TTL @param max_age
+   * to be sent to the downstream host.
+   * @return std::string the value of the new cookie
+   *
+   * marked const so it can be called from hashKey(), and because all mutated
+   * state is only visible when calling onUpstreamHeaders()
+   */
+  std::string addDownstreamSetCookie(const std::string& key, std::chrono::seconds max_age) const {
+    // The cookie value should be the same per connection so that if multiple
+    // streams race on the same path, they all receive the same cookie.
+    // Since the downstream port is part of the hashed value, multiple HTTP1
+    // connections can receive different cookies if they race on requests.
+    std::string value;
+    const Network::Connection* conn = downstreamConnection();
+    if (conn) {
+      value = conn->remoteAddress().asString() + conn->localAddress().asString();
+    } else {
+      //TODO(akonradi) talk to mattklein123 and figure out when this can happen
```

Comment:
I agree this case doesn't seem to make sense. Did you see this message (or change to ASSERT and experiment) when running any integration tests?

Reply:
The integration tests don't ever hit this. Some of the unit tests do but that can be fixed through better mocks.